packages/@aws-cdk/aws-codepipeline/test/stages.test.ts

```diff
@@ -134,5 +136,53 @@ nodeunitShim({
 
       test.done();
     },
+
+    'configure notification on stage level'(test: Test) {
+      const stack = new cdk.Stack();
+      const sourceArtifact = new codepipeline.Artifact();
+      const pipeline = new codepipeline.Pipeline(stack, 'Pipeline', {
+        stages: [
+          {
+            stageName: 'Source',
+            actions: [new FakeSourceAction({ actionName: 'Fetch', output: sourceArtifact })],
+          },
+          {
+            stageName: 'Build',
+            actions: [new FakeBuildAction({ actionName: 'Gcc', input: sourceArtifact })],
+          },
+        ],
+      });
+
+      const stage = pipeline.stages[0];
```

Comment:
Just assign the Stage to a local variable before adding it to the Pipeline, and you won't have to do these gymnastics.